v2_user_config.go

```diff
@@ -36,12 +36,6 @@ func (ad *V2AppDefinition) UnmarshalJSON(data []byte) error {
 
 	result := V2AppDefinition(adc)
 
-	// validate app definition without validation context. validation context is
-	// given on server side to additionally validate specific definitions.
-	if err := result.Validate(nil); err != nil {
```

Comment:
We have to make sure that in all places where this is used, we add validation code.

- CLI 
- App-Service (in those calls that accept a definition)